modules/button/react/lib/Button.tsx

```diff
@@ -34,9 +40,14 @@ export interface ButtonProps
    * Note: not displayed at `small` size
    */
   icon?: CanvasSystemIcon;
+  /**
+   * The alternative container type for the button. Uses Emotion's special `as` prop.
+   * Will render an `a` tag instead of a `button` when defined.
+   */
+  as?: 'a' | undefined;
 }
 
-const Button = ({
+const Button: ButtonOrAnchorComponent<ButtonProps, typeof ButtonVariant> = ({
```

Comment:
A downside to this approach is react-docgen gets very confused about this type signature and doesn't do any prop table. We can re-evaluate when we look at documentation solutions.